Replacing hardcoded values by configuration

OneEyeMaker opened the pull request:
**_WIP Project. This PR isn't finished right now._**
This is attempt to fix #925 and replace hardcoded values with configuration.

Shadows-of-Fire commented:
So the random formatting changes everywhere.... why exactly?

Shadows-of-Fire commented on src/main/java/de/ellpeck/actuallyadditions/mod/config/values/ConfigIntValues.java:
```diff
@@ -79,6 +81,22 @@
+        }
+        this.currentValue = value;
+    }
+
```
So this whole method here is useless, and should just be config.getInt and use the min/max values of that method.

OneEyeMaker replied:
```Configuration``` class (by Forge) uses minimum and maximum values just for displaying purpose. When it gets values - it ignores ones. Check methods ```getInt()```, ```getDouble()``` and so on. These methods *completely* ignore minimum and maximum values. That's because I added these checks.

Shadows-of-Fire replied:
Those methods do not ignore the bounds given to them.  `getInt` and `getFloat` have checks in place.  And well, apparently there is no `getDouble`

OneEyeMaker replied:
Please, show me, where is this internal check. (I checked whole ```Configuration``` and ```Property``` classes and didn't found; I found it only in GUI configuration classes)

Shadows-of-Fire commented on src/main/java/de/ellpeck/actuallyadditions/mod/config/values/ConfigDoubleValues.java:
```diff
@@ -31,6 +33,22 @@
+        }
+        this.currentValue = value;
+    }
+
```
Switch this entire thing to config.getDouble.

Shadows-of-Fire commented on src/main/java/de/ellpeck/actuallyadditions/mod/items/ItemTeleStaff.java:
```diff
@@ -42,8 +43,8 @@ public ItemTeleStaff(String name){
-                    int baseUse = 200;
-                    int use = baseUse+(int)(baseUse*pos.hitVec.distanceTo(new Vec3d(player.posX, player.posY+(player.getEyeHeight()-player.getDefaultEyeHeight()), player.posZ)));
+                    int baseUse = ConfigIntValues.TELEPORT_STAFF_ENERGY_USE.getValue();
+                    int use = baseUse + (int)(baseUse * pos.hitVec.distanceTo(new Vec3d(player.posX, player.posY + (player.getEyeHeight() - player.getDefaultEyeHeight()), player.posZ)));
```
Are you just randomly running formatting on methods for no reason?

Shadows-of-Fire commented on src/main/java/de/ellpeck/actuallyadditions/mod/util/StackUtil.java:
```diff
@@ -36,9 +36,14 @@ public static ItemStack validateCheck(ItemStack stack){
+    	if(i instanceof IDisableableItem){
+    	    return !((IDisableableItem) i).isDisabled();
+        }
         return !stack.isEmpty();
```
Hey look its you formatting my shit again for no reason...
You don't even touch this class.

OneEyeMaker replied:
This place - isn't my fault; there was mixed indenting (tabs and spaces). Replace to spaces?

Shadows-of-Fire commented on src/main/java/de/ellpeck/actuallyadditions/mod/items/ItemLeafBlower.java:
```diff
@@ -107,7 +108,8 @@ public boolean breakStuff(World world, int x, int y, int z){
 
-                    if(block != null && ((block instanceof BlockBush || block instanceof IShearable) && (this.isAdvanced || !block.isLeaves(world.getBlockState(pos), world, pos)))){
+                    if((block instanceof BlockBush || block instanceof IShearable) &&
+                       (this.isAdvanced || !block.isLeaves(world.getBlockState(pos), world, pos))){
```
Seriously

OneEyeMaker replied:
About moving imports on top and other formatting features. It'll better, if you post somewhere imported formatting configuration from IntelliJ IDEA (because most of the stuff occurs 'on the fly').

OneEyeMaker commented:
@Shadows-of-Fire 
Commits about Bio Reactor and Canola Press was made before your review.
In last commit I moved imports back and fixed indenting (little bit).
Check full changelog and say, what's wrong with formatting.

Ellpeck commented:
Yea I'm just stopping it right here.
I'm absolutely not a fan of the whole "put every little fucking thing into the config" thing, and I still have enough love for the mod to make this *not* happen.
